tools/spatialDraw/index.js

```diff
@@ -0,0 +1,324 @@
+sizeCircles.forEach((circle, i) => {
+    circle.setColor = (color) => {
+        Array.from(circle.children)[0].setAttribute('fill', color);
+    };
+    circle.addEventListener('pointerdown', e => {
+        e.stopPropagation();
+        const nextCircle = sizeCircles[(i + 1) % sizeCircles.length];
+        drawingManager.setSize(nextCircle.dataset.size);
+    });
+});
+const colorCircles = Array.from(ui.children).filter(child => child.classList.contains('color'));
+const eraseCircle = Array.from(ui.children).filter(child => child.classList.contains('erase'))[0];
+colorCircles.forEach(circle => {
+    circle.style.backgroundColor = circle.dataset.color;
+    circle.addEventListener('pointerdown', e => {
+        e.stopPropagation();
+        drawingManager.setColor(circle.dataset.color);
+    });
+});
+eraseCircle.addEventListener('pointerdown', e => {
+    e.stopPropagation();
+    colorCircles.forEach(colorCircle => colorCircle.classList.remove('active'));
+    eraseCircle.classList.add('active');
+    drawingManager.setEraseMode(true);
+});
+const undoCircle = Array.from(ui.children).filter(child => child.classList.contains('undo'))[0];
+undoCircle.addEventListener('pointerdown', e => {
+    e.stopPropagation();
+    undoCircle.classList.add('active');
+    setTimeout(() => undoCircle.classList.remove('active'), 150);
+    drawingManager.undoEvent();
+});
+const cursorMenuOptions = Array.from(document.querySelectorAll('.cursor'));
+cursorMenuOptions.forEach(cursorMenuOption => {
+    cursorMenuOption.addEventListener('pointerdown', e => {
+        e.stopPropagation();
+        drawingManager.setCursor(drawingManager.cursorMap[cursorMenuOption.dataset.cursor]);
+    });
+});
+
+const envelope = new Envelope(spatialInterface, [], uiParent, launchButton, false, false);
+envelope.onOpen(() => {
+    if (!rendererStarted) {
+        initRenderer().then(() => {
+            initDrawingApp();
+            appActive = true;
+            scene.visible = true;
+        });
+    } else {
+        drawingManager.enableInteractions();
+        appActive = true;
+        scene.visible = true;
+    }
+});
+envelope.onClose(() => {
+    drawingManager.disableInteractions();
+    appActive = false;
+    scene.visible = false;
+    location.reload();
+});
+
+function resizeText() {
+    text.innerText = text.innerText.toUpperCase();
+    const fontSize = Math.min(70, (((text.innerText.length * 7) + 500) / (text.innerText.length))); // font size increases up to 45pt
+    text.style.fontSize = fontSize + 'pt';
+}
+resizeText();
+
+function resetScroll() {
+    if (window.scrollX !== 0 || window.scrollY !== 0) {
+        window.scrollTo(0, 0); // don't let keyboard events scroll the window
+    }
+    parent.postMessage(JSON.stringify({resetScroll: true}), '*');
+}
+resetScroll();
+
+function initDrawingApp() {
+    drawingManager = new DrawingManager(mainContainerObj, camera);
```

Comment:
I like the way you structured this with the DrawingManager! Nice work managing the potential complexity of this system.